<?xml version="1.0" ?>

<launch>

  <!-- Arguments -->
  <arg name="arduino_ip" default="192.168.1.3" doc="IP of the Arduino on the local network."/>
  <arg name="task_path" default="$(find teaching_interface_ros)/tasks/" doc="Path to the taught task configuration." />
  <arg name="task_name" doc="Task name." />

  <!-- Launch teaching interface client -->
  <node name="teaching_interface_client" pkg="teaching_interface_ros" type="teaching_interface_client" output="screen" >
    <param name="arduino_ip" value="$(arg arduino_ip)" />
  </node>

  <!-- Launch teaching node -->
  <node name="teach_task" pkg="teaching_interface_ros" type="teach_task" respawn="false" output="screen" >
    <param name="path" type="str" value="$(arg task_path)$(arg task_name).yaml"/>
    <rosparam command="load" file="$(find teaching_interface_ros)/config/ur10_servo_config.yaml" />
  </node>

  <!-- Launch UR10 Driver -->
  <include file="$(find ur_modern_driver)/launch/ur10_ros_control.launch" >
    <arg name="robot_ip" value="192.168.1.32" />
    <arg name="controllers" default="joint_state_controller force_torque_sensor_controller joint_group_vel_controller"/>
    <arg name="stopped_controllers" default="pos_based_pos_traj_controller vel_based_pos_traj_controller"/>
  </include>

  <!-- Launch UR10 MoveIt -->
  <include file="$(find ur10limited_moveit_config)/launch/move_group.launch">
    <arg name="publish_monitored_planning_scene" value="true" />
  </include>
  <!-- <include file="$(find ur10limited_moveit_config)/launch/moveit_rviz.launch" >
    <arg name="rviz_config" value="$(find ur10limited_moveit_config)/launch/moveit.rviz" />
  </include> -->

  <!-- Launch servo node -->
  <node name="servo_node" pkg="teaching_interface_ros" type="servo_node" output="screen" >
    <param name="publish_rate" type="int" value="100" />
    <param name="acceleration_limit_trans" type="double" value="2.0" />
    <param name="acceleration_limit_rot" type="double" value="2.0" />
    <param name="parameter_ns" type="string" value="ur10_servo" />
    <rosparam ns="ur10_servo" command="load" file="$(find teaching_interface_ros)/config/ur10_servo_config.yaml" />
    <remap from="/servo_node/teaching_command" to="/teaching_interface_client/teaching_command" />
  </node>
  
</launch>
